tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -542,6 +542,41 @@ public synchronized void remove() throws IOException {
     }
   }
 
+  /**
+   * Removes the eldest {@code n} elements.
+   *
+   * @throws java.util.NoSuchElementException if the queue is empty
+   */
+  public synchronized void remove(int n) throws IOException {
+    if (isEmpty()) throw new NoSuchElementException();
+    if (n == 1) {
+      remove();
+    } else if (n >= elementCount) {
```

Comment:
Including greater than here is interesting behavior. This means that a loop of 10 `remove(1)` vs. `remove(10)` won't have the same behavior. I think we should throw in the greater than case.